great_expectations/core/expectation_suite.py

```diff
@@ -124,6 +128,52 @@ def add_citation(
         )
         self.meta["citations"].append(citation)
 
+    def update_last_citation(
```

Comment:
We want to update the most recent citation using `data_context.get_validator` instead of adding a new citation every time `get_validator` is called.